packages/flutter/lib/src/material/input.dart

```diff
@@ -137,6 +147,15 @@ class _InputState extends State<Input> {
       config.onSubmitted(_value);
   }
 
+  void _handleSelectionChanged(TextSelection selection) {
+    if (_isAttachedToKeyboard) {
+      _keyboardHandle.setSelection(selection.start, selection.end);
+    } else {
+      _editableString.setSelection(selection);
+      _requestKeyboard();
+    }
+  }
```

Comment:
this function is confusing, in part because it calls _requestKeyboard in a branch which guarantees that _requestKeyboard only does a one line request. I think rather than factoring _requestKeyboard out, we should just factor out the `Focus.moveTo(config.key);` line.